### PR TITLE
Make bundling possible for easier app runs

### PR DESCRIPTION
**Pull request by anthony2261**
*Issue #, if available:*
#126 

*Description of changes:*
- create dockerfile for bundling for Windows using Wine
- run migrations on app startup if is running from bundled app
- mount frontend assets and serve from backend iff bundled
- Add instructions to run bundling on 3 major OS's
- fix online migrations

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by anthony2261** [comment]
@RamiAwar can you check the `BUNDLING.md` file for instructions on bundling for MacOS (don't forget to run the frontend build commands at the top of the file first) and check if the bundling works well for MacOS?

**Comment by RamiAwar** [comment]
> You should be able to run the executable called `main` under `macos_dist/main/main` at the root of the repo and get the frontend once you open localhost:7377

Why 7377 not 2222 like the docker file?

File: text2sql-backend/main.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 import uvicorn
 from alembic import command
 from alembic.config import Config
+import webbrowser
```
Comment (RamiAwar) [inline text2sql-backend/main.py:10]: woahh this is built-in??

Comment (anthony2261) [reply text2sql-backend/main.py]: Apparently so hahahah

**Comment by anthony2261** [comment]
> > You should be able to run the executable called `main` under `macos_dist/main/main` at the root of the repo and get the frontend once you open localhost:7377
> 
> Why 7377 not 2222 like the docker file?

@RamiAwar Don't know why I can't directly reply to this comment but whatever, it's because that's the backend server's port. If we tell it to run on port 2222, we would need to tell the frontend to call the backend on port 2222 instead of 7377 (which is easily doable), see `text2sql-frontend/src/services/api_client.ts` for how we're specifying the port. Let me know we should do it on port 2222, I'll make the changes

**Review by RamiAwar — APPROVED** [review]
Can you also update the README on how we should run the app locally now? 

How do frontend changes get synced if it's getting served through fastapi? How do we build + serve through fastapi? Not documented yet

**Comment by RamiAwar** [comment]
One thing still missing here is 1) conditionally running this only when we want to release and 2) creating a github release out of these artifacts. Upload artifacts are only useful for stuff needs to stay in the build. In this case we're doing a full release.

But let's do that in other PRs.

**Pull request merged**